x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/ParsingTests.java

```diff
@@ -205,16 +213,145 @@ public void testInvalidSample() {
         );
     }
 
-    private String error(String query, QueryParams params) {
-        ParsingException e = expectThrows(
-            ParsingException.class,
-            () -> defaultAnalyzer.analyze(parser.createStatement(query, params, TEST_CFG))
+    public void testSet() {
+        EsqlQuery query = parse("SET foo = \"bar\"; row a = 1", new QueryParams());
+        assertThat(query.plan(), is(instanceOf(Row.class)));
+        assertThat(query.settings().size(), is(1));
+        assertThat(query.settings().get(0).fields().size(), is(1));
+        assertThat(query.settings().get(0).fields().get(0).name(), is("foo"));
+        assertThat(query.settings().get(0).fields().get(0).child().fold(FoldContext.small()), is(BytesRefs.toBytesRef("bar")));
```

Comment:
This looks a bit too complex with 2 nested lists. Is it possible to collect query settings into a flat structure?
I imagine usage could become much simpler if we have something like `assertThat(query.settings().setting("foo").fold(..), is(..))`